get_target_property() called with non-existent target gRPC::grpc_cpp_plugin

Trying to get etcd-cpp-apiv3  in OpenBMC distribution.  Seeing below build failure.

xxxx/etcd-cpp-apiv3/build$ cmake .. -DCPPREST_EXCLUDE_WEBSOCKETS=ON
-- Toolchain file defaulted to 'xxxx/sysroots/x86_64-oesdk-linux/usr/share/cmake/OEToolchainConfig.cmake'
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: xxxx/sysroots/x86_64-oesdk-linux/usr/bin/arm-openbmc-linux-gnueabi/arm-openbmc-linux-gnueabi-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: xxxx/sysroots/x86_64-oesdk-linux/usr/bin/arm-openbmc-linux-gnueabi/arm-openbmc-linux-gnueabi-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'Release' as none was specified.
-- Building etcd-cpp-apiv3 with C++11
-- Found Boost: xxxx/sysroots/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/usr/lib/cmake/Boost-1.81.0/BoostConfig.cmake (found version "1.81.0") found components: system thread random 
-- Found OpenSSL:xxxx/sysroots/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/usr/lib/libcrypto.so (found version "3.0.7")  
CMake Warning at xxxx/sysroots/x86_64-oesdk-linux/usr/share/cmake-3.24/Modules/FindProtobuf.cmake:524 (message):
  Protobuf compiler version 3.6.1 doesn't match library version 3.21.5
Call Stack (most recent call first):
  CMakeLists.txt:110 (find_package)


-- Found Protobuf: xxxx/sysroots/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/usr/lib/libprotobuf.so (found version "3.21.5") 
-- Found ZLIB: xxxx/sysroots/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/usr/lib/libz.so (found version "1.2.13") 
-- Found c-ares: xxxx/sysroots/armv7ahf-vfpv4d16-openbmc-linux-gnueabi/usr/lib/cmake/c-ares/c-ares-config.cmake (found version "1.18.0") 
-- Found Threads: TRUE  
-- Found RE2 via CMake.
CMake Error at CMakeLists.txt:124 (get_target_property):
  get_target_property() called with non-existent target
  "gRPC::grpc_cpp_plugin".


-- Found GRPC: gRPC::gpr;gRPC::grpc;gRPC::grpc++,  (found version: "1.53.0")
-- Performing Test W_NO_CPP17_EXTENSIONS
-- Performing Test W_NO_CPP17_EXTENSIONS - Success
-- Configuring incomplete, errors occurred!
See also "xxxx/etcd-cpp-apiv3/build/CMakeFiles/CMakeOutput.log".

Looking for help on debugging this failure.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -119,7 +119,7 @@ if(Protobuf_PROTOC_EXECUTABLE)
 endif()
 
 find_package(gRPC QUIET)
-if(gRPC_FOUND)
+if(gRPC_FOUND AND TARGET gRPC::grpc AND TARGET gRPC::grpc_cpp_plugin)
     set(GRPC_LIBRARIES gRPC::gpr gRPC::grpc gRPC::grpc++)
     get_target_property(GRPC_CPP_PLUGIN gRPC::grpc_cpp_plugin LOCATION)
     get_target_property(GRPC_INCLUDE_DIR gRPC::grpc INTERFACE_INCLUDE_DIRECTORIES)

